source/_components/switch.xiaomi_vacuum.markdown

```diff
@@ -0,0 +1,59 @@
+---
+layout: page
+title: "Xiaomi Mi Robot Vacuum"
+description: "Instructions how to integrate your Xiaomi Mi Robot Vacuum within Home Assistant."
+date: 2017-05-05 18:11
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: xiaomi_vacuum.png
+ha_category: Switch
+ha_release: 0.47
+---
+
+The `xiaomi_vacuum`switch platform allows you to control the state of your [Xiaomi Mi Robot Vacuum](http://www.mi.com/roomrobot/). 
+Current supported features are start and stop (goes to dock).
+
+{% linkable_title Getting started %}
+
+To be able to communicate with the vacuum one needs to have its IP address as well as an encryption token.
+This token is only attainable before the device has been connected over the app to your local wifi.
+
+<p class='note warning'>
+If your Home Assistant installation is running on [Virtualenv](/docs/installation/virtualenv/#upgrading-home-assistant), make sure you activate it by running the commands below.</p>
+
+```bash 
+$ sudo su -s /bin/bash homeassistant
+$ source /srv/homeassistant/bin/activate
+```
+
+In order to fetch the token, reset the robot, connect to its the network its announcing (rockrobo-XXXX)
+and run the following command:
+
+```bash
+mirobo discover
```

Comment:
`In order to fetch the token, reset the vacuum following the instructions in the manual. Then,
connect to its built-in wireless network (**rockrobo-XXXX**) with a computer that has the
[python-mirobo](https://github.com/rytilahti/python-mirobo) package installed.
Run the following command:`